aspnetcore/migration/webapi.md

```diff
@@ -4,19 +4,187 @@ author: ardalis
+
+ASP.NET Core 6.0 projects enable nullable reference types by default. Nullable reference types are enabled or disabled for the entire project with the following project files property:
+
+```xml
+<PropertyGroup>
+    <Nullable>enable</Nullable>
+</PropertyGroup>
+```
+
+For more information, see [Nullable reference types](/dotnet/csharp/nullable-references).
+
+Examine *Models/Product.cs*. With nullable reference types enabled for the ProductCore project, helpful `non-nullable property` warnings are visible for the `Name` and `Category` properties.
+
+![Nullable reference type warning](webapi/_static/non-nullable-warning.png)
+
+In this case, the intent is that the `Name` and `Category` can be nullable types.
+
+1. Add the `?` annotation to declare the `Name` and `Category` properties as nullable types:
+
+```diff
+public int Id { get; set; }
+- public string Name { get; set; }
++ public string? Name { get; set; }
+- public string Category { get; set; }
++ public string? Category { get; set; }
+public decimal Price { get; set; }
+```
+
+## Migrate the controller
+
+Update the `ProductsController` for ASP.NET Core:
+
+The following code shows the updated ASP.NET Core `ProductsController`:
+
+[!code-csharp[](webapi/sample/6.x/ProductsCore/Controllers/ProductsController.cs?highlight=1,2,4,8,33,50,57)]
+
+1. Copy *Controllers/ProductsController.cs* and the *Models* folder from the original project to the new one.
+1. Change the copied files' root namespace to `ProductsCore`.
+1. Update the `using ProductsApp.Models;` statement to `using ProductsCore.Models;`.
+
+The following components don't exist in ASP.NET Core:
+
+* `ApiController` class
+* `System.Web.Http` namespace
+* `IHttpActionResult` interface
+
+Make the following changes:
+
+1. Change `ApiController` to <xref:Microsoft.AspNetCore.Mvc.ControllerBase>. Add `using Microsoft.AspNetCore.Mvc;` to resolve the `ControllerBase` reference.
+1. Delete `using System.Web.Http;`.
+1. Change the `GetProduct` action's return type from `IHttpActionResult` to `ActionResult<Product>`.
+1. Simplify the `GetProduct` action's `return` statement to the following statement:
+
+    ```csharp
+    return product;
+    ```
+
+## Configure routing
+
+The following code code in the App_Start/WebAPIConfig.cs added an endpoint for the ASP.NET Core 3.x ProductsApp app:
+
+[!code-csharp[](webapi/sample/3.x/ProductsApp/App_Start/WebAPIConfig.cs?highlight=15, 17-21)]
+
+ASP.NET Core 6.0 provides a minimal hosting model in which the endpoint routing middleware wraps the entire middleware pipeline, therefore routes can be added directly to the <xref:Microsoft.AspNetCore.Builder.WebApplication> without an explicit call to <xref:Microsoft.AspNetCore.Builder.EndpointRoutingApplicationBuilderExtensions.UseEndpoints%2A> or <xref:Microsoft.AspNetCore.Builder.EndpointRoutingApplicationBuilderExtensions.UseRouting%2A> to register routes.
```

Comment:
We almost never put version numbers in the doc. This is the V6 version and won't be updated when V7 is released.

Reply:
@Rick-Anderson .   It is helpful I think to know when the switch was made.  Perhaps something like "ASP.NET Core 6.0 introduced a minimal hosting model..." ?

Reply:
No, doesn't meet the bar. If they're on V6, we don't care about history.

Reply:
I disagree.  This is a migration doc, not just any doc.  I will make the change anyway, but I think it needed to be explicit.